osh/builtin_process.py

```diff
@@ -69,20 +69,48 @@ def __init__(self, job_control, job_list, waiter):
     def Run(self, cmd_val):
         # type: (cmd_value.Argv) -> int
 
-        pid = self.job_list.GetLastStopped()
-        if pid == -1:
+        job_spec = '' # get current job by default
+        if len(cmd_val.argv) > 1:
+            job_spec = cmd_val.argv[1]
+
+        job = self.job_list.GetJobWithSpec(job_spec)
+        if job is None:
             log('No job to put in the foreground')
             return 1
 
+        pgid = job.ProcessGroupId()
         # TODO: Print job ID rather than the PID
-        log('Continue PID %d', pid)
+        log('Continue PID %d', pgid)
         # Put the job's process group back into the foreground. GiveTerminal() must
         # be called before sending SIGCONT or else the process might immediately get
         # suspsended again if it tries to read/write on the terminal.
-        pgid = posix.getpgid(pid)
         self.job_control.MaybeGiveTerminal(pgid)
+        job.SetForeground()
+        # needed for Wait() loop to work
+        job.state = job_state_e.Running
         posix.killpg(pgid, SIGCONT)
-        return self.job_list.WhenContinued(pid, self.waiter)
+
+        status = 1 # error
```

Comment:
hm it seems like this should always be overwritten?

we can initialize it to -1 if we need that for mycpp ?

Reply:
Ah yes looks like it. I copied this from the `Wait` builtin below where the logic is a bit more compilcated. Will set to `-1`